### ISSUE TITLE
Error while running extract_data

### ISSUE THREAD

**Issue by jel0624** [int-brain-lab/iblapps#127]
I encountered this error while running extract_data:

![Screenshot 2024-10-01 163741](https://github.com/user-attachments/assets/a7bccb9f-ca90-42a8-b86d-90ae36426931)

I was wondering if this is a problem???

Best,
Jay

**Comment by jel0624** [comment]
I tried nevertheless to run the IBL GUI. I get an error like the following:
![Screenshot 2024-10-01 165044](https://github.com/user-attachments/assets/da764c23-833e-4a06-ab97-266ee13db696)

**Comment by mayofaulkner** [comment]
Hi Jay,

For the first issue could you run the following code and paste the output here
```python
import numpy as np
from pathlib import Path
out_path = Path('') # This should be the same out_path used above in the extraction

attributes = ['clusters', 'amps', 'times', 'depths']
for att in attributes:
    data = np.load(out_path.joinpath(f'spikes.{att}.npy'))
    print(f'spikes.{att}: {data.shape}, {np.nanmin(data)}, {np.nanmin(data)}')

data = np.load(out_path.joinpath('clusters.channels.npy'))
print(f'clusters.channels: {data.shape}')
```

For the second issue can you expand on how you got your xyz_picks.json. Did you make sure to transform it to the IBL coordinates?

**Comment by jel0624** [comment]
![Screenshot 2024-10-08 153200](https://github.com/user-attachments/assets/88d6c5c4-d818-43b7-bd63-5e126cd9207d)

for the second issue, I obtained probe_ccf.npy from [AP_histology](https://github.com/petersaj/AP_histology)
and used the following code to transform it:

`import numpy as np
from pathlib import Path
import json
from ibllib.atlas import AllenAtlas

atlas = AllenAtlas(25)

brainreg_path = Path(r'C:\Neuropixels_sorting_folder\JL101_20240914_g0\JL101_1_processed\probe_ccf.npy')
# Load in coordinates of track in CCF space (order - apdvml, origin - top, left, front voxel
xyz_apdvml = np.load(brainreg_path)
# Convert to IBL space (order - mlapdv, origin - bregma)
xyz_mlapdv = atlas.ccf2xyz(xyz_apdvml, ccf_order='apdvml') * 1e6
xyz_picks = {'xyz_picks': xyz_mlapdv.tolist()}

# Path to save the data (same folder as where you have all the data)
output_path = Path(r'C:\Neuropixels_sorting_folder\JL101_20240914_g0\alf')
with open(Path(output_path, 'xyz_picks.json'), "w") as f:
    json.dump(xyz_picks, f, indent=2)`

**Comment by mayofaulkner** [comment]
Hi,

Thanks for the response. Is there any chance you could share your ks output and also the proce_ccf.npy file that you have. On dropbox or gdrive? I can then take a look.

Many thanks!

**Comment by jel0624** [comment]
https://www.dropbox.com/scl/fo/tvhzysf536usxcf8ocnpx/AODPA7RBbbH76hQHtgv2ZnA?rlkey=xxzaxfc4bc6whtpkb2kf3jd8h&st=fmbu84rn&dl=0

**Comment by mayofaulkner** [comment]
Hi Jay,

Thanks for the data. For the first issue please can you do `pip install ibllib --upgrade` and run the extraction code again. 

For the second issue about the location of the histology. There are two things, the first is that the data you have is in pixel coordinates so you need to multiply by the resolution of the atlas used by AP-histology (10 um) to convert to um and the second is that it seems the ordering is mldvap

The following code should be get you the correct xyz_picks

```
import numpy as np
from pathlib import Path
import json
from iblatlas.atlas import AllenAtlas

atlas = AllenAtlas(25)

brainreg_path = Path(r'C:\Neuropixels_sorting_folder\JL101_20240914_g0\JL101_1_processed\probe_ccf.npy')
picks = np.load(brainreg_path)
# Convert picks to um by multiplying by resolution of atlas
picks = picks * 10
# Swap axis to picks has order apdvml (current oder is a mldvap)
picks[:, [2, 0]] = picks[:, [0, 2]]

# Convert to mlapdv relative to bregma
xyz_mlapdv = atlas.ccf2xyz(picks, ccf_order='apdvml') * 1e6
xyz_picks = {'xyz_picks': xyz_mlapdv.tolist()}

output_path = Path(r'C:\Neuropixels_sorting_folder\JL101_20240914_g0\alf')
with open(Path(output_path, 'xyz_picks.json'), "w") as f:
    json.dump(xyz_picks, f, indent=2)
```

When I load it in the GUI i get the probe in this location, does it look reasonable? 

<img width="338" alt="Screen Shot 2024-10-16 at 11 27 40 AM" src="https://github.com/user-attachments/assets/aab09296-3ca3-499c-aa1c-1f66c4370b9c">

**Comment by jel0624** [comment]
Thanks for your help!
The coordinates should be around OFC, and that seems to be around GPe(?)
maybe some axis got flipped?

**Comment by mayofaulkner** [comment]
Does this look more reasonable?

<img width="341" alt="Screen Shot 2024-10-16 at 4 43 06 PM" src="https://github.com/user-attachments/assets/8dc5f9d3-eaf9-40fa-abfd-32a49ee168fd">

In that case you were correct with the apdvml order. All you need to do is multiply by 10 to ensure the coordinates are in um.
```
import numpy as np
from pathlib import Path
import json
from iblatlas.atlas import AllenAtlas

atlas = AllenAtlas(25)

brainreg_path = Path(r'C:\Neuropixels_sorting_folder\JL101_20240914_g0\JL101_1_processed\probe_ccf.npy')
picks = np.load(brainreg_path)
# Convert picks to um by multiplying by resolution of atlas
picks = picks * 10

# Convert to mlapdv relative to bregma
xyz_mlapdv = atlas.ccf2xyz(picks, ccf_order='apdvml') * 1e6
xyz_picks = {'xyz_picks': xyz_mlapdv.tolist()}

output_path = Path(r'C:\Neuropixels_sorting_folder\JL101_20240914_g0\alf')
with open(Path(output_path, 'xyz_picks.json'), "w") as f:
    json.dump(xyz_picks, f, indent=2)
```

**Comment by jel0624** [comment]
Yes, this is the coordinates!! Thanks for your help, and will give this a try now

**Comment by jel0624** [comment]
![Screenshot 2024-10-22 110527](https://github.com/user-attachments/assets/fae6af1c-7aed-420b-b87c-56a030229709)
Unfortunately, I still got this error while trying to extract the data.

**Comment by jel0624** [comment]
I fixed it by changing the _phy_spikes_subset.waveforms.npy to _phy_spikes_subset_waveforms.npy

For some reason, python didn't like the double dot notation. 
Thanks!
